superset/thumbnails/digest.py

```diff
@@ -0,0 +1,85 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from flask import current_app
+
+from superset.tasks.types import ExecutorType
+from superset.tasks.utils import get_executor, get_initiator
+from superset.utils.hashing import md5_sha_from_str
+
+if TYPE_CHECKING:
+    from superset.models.dashboard import Dashboard
+    from superset.models.slice import Slice
+
+logger = logging.getLogger(__name__)
+
+
+def _adjust_string_for_executor(
+    unique_string: str,
+    executor_type: ExecutorType,
+    executor: str,
+) -> str:
```

Comment:
If the API endpoints for screenshots provide enough security, we probably don't need this function anymore.

Reply:
I agree, I'll simplify this.